### DISCUSSION TITLE
trouble converting wpg files

### DISCUSSION THREAD

**Discussion by mikey245** [ImageMagick/ImageMagick#7864] [Help]
### ImageMagick version

ImageMagick-7.1.1-Q16-HDRI

### Operating system, version and so on

Windows 10 for Workstations

### Description

I have a lot of wpg files dating from the 90’s and early 2000’s that I would like to convert to svg files.   For example, one file is afghan.wpg.  Command magick afghan.wpg afghan.svg produces a response of:   magick.exe: image file does not contain any image data `afghan.wpg' @ error/wpg.c/ReadWPGImage/1632.  There’s nothing wrong with the wpg file; Presentations and Inkscape can open it with no trouble.  If I try converting one of the example files from the imagemagick website, like tru256.wpg, magick.exe works OK.  Must be some variations in the wpg format, and Imagemagick can handle only some of them.  Any solution?

**Comment by snibgo** [comment]
You say that IM can't read some WPG files, but you haven't linked to them, or zipped them and uploaded the zip file.

In a quick read of coders/wpg.c I can't see where IM would read and process vector data.

**Comment by mikey245** [comment]
OK.  I've uploaded a couple of wpg files.  afgahn.wpg, which magick.exe can't process, and tru256.wpg, which magick.exe can process.
[wpg_files.zip](https://github.com/user-attachments/files/18262998/wpg_files.zip)

**Comment by fmw42** [comment]
I can open TRU256.WPG and convert to PNG, but for AFGHAN.WPG, I get

`magick: image file does not contain any image data `AFGHAN.WPG' @ error/wpg.c/ReadWPGImage/1632.`

I suspect that the latter does not fully conform to the WPG specs and IM is very strict about that.

I am on IM 7.1.1.43 Mac OSX Ventura

**Comment by snibgo** [comment]
afghan.wpg contains entirely vector data, which IM can't read.

IM is a raster image processor, with limited vector capabilities. You want to convert WPG to SVG, so I assume you want to keep them as vectors. I suggest using Inkscape for this task. Inkscape can be used from the command line, so you could convert all your WPG files in batch.

**Comment by mikey245** [comment]
OK. Thanks for the info.  Suggest that the limitation of IM for wpg files be stated as a comment next to WPG on the webpage (imagemagick.org/script/formats.php) that lists the formats that IM can process.  I'll see if I can understand the command line options for Inkscape.  I tried a package called InkscapeBatchConvert that uses the command line.  It went through the motions with no error messages but produced no output.
